src/Loess.jl

```diff
@@ -179,48 +195,141 @@ end
     y₂, dy₂ = model.predictions_and_gradients[[v₂]]
 
     b_int = cubic_interpolation(v₁, y₁, dy₁, v₂, y₂, dy₂)
 
-    return evalpoly(z, b_int)
+    return evalpoly(x, b_int)
+end
+
+struct LoessPrediction{T}
+    predictions::Vector{T}
+    lower::Vector{T}
+    upper::Vector{T}
+    sₓ::Vector{T}
+    δ₁::T
+    δ₂::T
+    s::T
+    ρ::T
 end
 
-function predict(model::LoessModel, zs::AbstractVector)
+"""
+    predict(
+        model::LoessModel,
+        x::AbstractVecOrMat = modelmatrix(model);
+        interval::Union{Symbol, Nothing}=nothing,
+        level::Real=0.95
+    )
+
+Compute predictions from the fitted Loess `model` at the values in `x`.
+By default, only the predictions are returned. If `interval` is set to
+:confidence` then a `LoessPrediction` struct is returned which contains
+upper and lower confidence bounds at `level` as well the the quantities
+used for computing the confidence bounds.
+
+When the prodictor takes values at the edges of the KD tree, the predictions
+are already computed and the function is simply a look-up. For other values
+of the preditor, a cubic spline approximation is used for the prediction.
+
+When confidence bounds are requested, a matrix of size ``n \times n`` is
+constructed where ``n`` is the length of the fitted data vector. Hence,
+this caluculation is only feasible when ``n`` is not too large. For details
+on the calculations, see the cited reference.
+
+Reference: Cleveland, William S., and Eric Grosse. "Computational methods
+for local regression." Statistics and computing 1, no. 1 (1991): 47-62.
+"""
+function predict(
+    model::LoessModel,
+    x::AbstractVecOrMat = modelmatrix(model);
+    interval::Union{Symbol, Nothing}=nothing,
+    level::Real=0.95
+)
     if size(model.xs, 2) > 1
         throw(ArgumentError("multivariate blending not yet implemented"))
     end
 
-    return [predict(model, z) for z in zs]
-end
+    if interval !== nothing && interval !== :confidence
+        if interval === :prediction
+            throw(ArgumentError("predictions not implemented. If you know how to compute them then please file an issue and explain how."))
+        end
+        throw(ArgumentError(LazyString("interval must be either :prediction or :confidence but was :", interval)))
+    end
 
-function predict(model::LoessModel, zs::AbstractMatrix)
-    if size(model.xs, 2) != size(zs, 2)
-        throw(DimensionMismatch("number of columns in input matrix must match the number of columns in the model matrix"))
+    if !(0 < level < 1)
+        throw(ArgumentError(LazyString("level must be between zero and one but was ", level)))
     end
 
-    if size(zs, 2) == 1
-        return predict(model, vec(zs))
+    predictions = [predict(model, _x) for _x in x]
+
+    if interval === nothing
+        return predictions
     else
-        return [predict(model, row) for row in eachrow(zs)]
+        # see Cleveland and Grosse 1991 p.50.
+        L = hatmatrix(model)
+        L̄ = L - I
```

Comment:
Here you could also save a few allocations:

```suggestion
        for i in diagind(L)
            L[i] -= 1
        end
        L̄ = L
```